daemon/k8s_watcher.go

```diff
@@ -476,6 +476,15 @@ func (d *Daemon) endpointAddFn(obj interface{}) {
 			return
 		}
 	}
+
+	d.policy.Mutex.Lock()
+	defer d.policy.Mutex.Unlock()
+	if err := d.policy.DeleteEndpointGeneratedEgressRules(svcns, *newSvcEP); err != nil {
```

Comment:
This makes me uncomfortable in the sense you are making policy change on all nodes whenever an endpoint appears on the cluster.